app/com/arpnetworking/metrics/portal/scheduling/JobExecutorActor.java

```diff
@@ -74,23 +74,58 @@
     private boolean _currentlyExecuting = false;
     private Optional<CachedJob<T>> _cachedJob = Optional.empty();
     private Optional<Instant> _nextRun = Optional.empty();
+    private final JobRefSerializer _refSerializer;
+
+    private JobExecutorActor(
+            final Injector injector,
+            final Clock clock,
+            final PeriodicMetrics periodicMetrics,
+            final JobRefSerializer refSerializer
+    ) {
+        _injector = injector;
+        _clock = clock;
+        _periodicMetrics = periodicMetrics;
+        _refSerializer = refSerializer;
+    }
 
     /**
      * Props factory.
      *
      * @param injector The Guice injector to use to load the {@link JobRepository} referenced by the {@link JobRef}.
      * @param clock The clock the scheduler will use, when it ticks, to determine whether it's time to run the next job(s) yet.
      * @param periodicMetrics The {@link PeriodicMetrics} that this actor will use to log its metrics.
+     * @param refSerializer The JobRefSerializer that this actor will use reconstruct its JobRef at startup.
      * @return A new props to create this actor.
      */
-    public static Props props(final Injector injector, final Clock clock, final PeriodicMetrics periodicMetrics) {
-        return Props.create(JobExecutorActor.class, () -> new JobExecutorActor<>(injector, clock, periodicMetrics));
+    public static Props props(
+            final Injector injector,
+            final Clock clock,
+            final PeriodicMetrics periodicMetrics,
+            final JobRefSerializer refSerializer
+    ) {
+        return Props.create(JobExecutorActor.class,
+                () -> new JobExecutorActor<>(injector, clock, periodicMetrics, refSerializer));
     }
 
-    private JobExecutorActor(final Injector injector, final Clock clock, final PeriodicMetrics periodicMetrics) {
-        _injector = injector;
-        _clock = clock;
-        _periodicMetrics = periodicMetrics;
+    @Override
+    public void preStart() throws Exception {
+        super.preStart();
+
+        final String actorName = getSelf().path().name();
+        final Optional<JobRef<?>> ref = _refSerializer.entityIDtoJobRef(actorName);
+        if (ref.isPresent()) {
+            final JobRef<T> castRef = unsafeJobRefCast(ref.get());
+            LOGGER.info()
+                    .setMessage("inferred job ref from name, triggering reload")
+                    .addData("jobRef", castRef.toString())
+                    .log();
+            reload(new Reload.Builder<T>().setJobRef(castRef).build());
```

Comment:
Unfortunately this adds another reload to the already excessive reload messages at startup because now the actor will send itself a reload message when it starts, if it can.

If there's some way I'm missing to avoid this and distinguish actor start from an actual reload, I'd be open to changing it.

Reply:
Is it safe to call `reload` in `preStart` vs sending a reload message to itself?

Given this new scheme for job actors, why do we need to send a reload message on actor creation at all? It seems completely redundant. Right?

If we go that route, the warn below likely should be an error and kill the actor (send msg).

Reply:
> Is it safe to call reload in preStart vs sending a reload message to itself?

I believe so, but I can change it to a `tell` just to be consistent.

> Given this new scheme for job actors, why do we need to send a reload message on actor creation at all? It seems completely redundant. Right?

I'm not sure if you're referring to the message from `preStart` or from the `JobCoordinator` when you refer to "send a reload message on actor creation".

We need to send it on pre-start because post-move we need to reinitialize our CachedJob and start the timer again.
As for JobCoordinator, I would agree that both reloading and starting the actor would be superfluous. Really, after these changes, all we care about is actor identity that we infer from the `JobRef`. This means externally we would ideally have two interactions:
* We only want the actor to start, we don't care about reload.
* We *actually* want the JobExecutorActor to reload its state, maybe a database update just happened or anti-entropy is running.

I can't see a way to tease these apart for anti-entropy, however, since from its POV there's no difference between restarting a stopped actor and asking it to reload.

Reply:
Yeah, I'm fine with reloads happening in two cases:

* Underlying data change
* Anti-entropy

However, the reload on actor start should be removed.